File: bench/words.py
WORDS = 2000000

def main():
    counts = {}
    seed = 42
    total = 0
    for _ in range(WORDS):
        seed = (seed * 6364136223846793005 + 1442695040888963407) & 0xFFFFFFFFFFFFFFFF
        k = (seed >> 33) % 50000
        word = "w%d" % k
        counts[word] = counts.get(word, 0) + 1
        total += 1
    print(len(counts), total)

main()

File: bench/words.go
package main

import "fmt"

const words = 10000000

func main() {
	counts := make(map[string]int64)
	var seed uint64 = 42
	var total int64
	for i := 0; i < words; i++ {
		seed = seed*6364136223846793005 + 1442695040888963407
		k := (seed >> 33) % 50000
		word := fmt.Sprintf("w%d", k)
		counts[word]++
		total++
	}
	fmt.Println(len(counts), total)
}
